### PR TITLE
Code quality/increase coverage of default device response

### PR DESCRIPTION
**Pull request by arjen0203**

### CODE REVIEW

**Review by mvanOsch — APPROVED** [review]
good

File: osgp/platform/osgp-adapter-domain-core/src/test/java/org/opensmartgridplatform/adapter/domain/core/application/services/DefaultDeviceResponseServiceTest.java [hunk 1]
```diff
@@ -0,0 +1,139 @@
+package org.opensmartgridplatform.adapter.domain.core.application.services;
```
Comment (robindenadel) [inline osgp/platform/osgp-adapter-domain-core/src/test/java/org/opensmartgridplatform/adapter/domain/core/application/services/DefaultDeviceResponseServiceTest.java:1]: licence header is missing

Comment (robindenadel) [inline osgp/platform/osgp-adapter-domain-core/src/test/java/org/opensmartgridplatform/adapter/domain/core/application/services/DefaultDeviceResponseServiceTest.java:1]: indentation is not ok, check your IDE settings

File: osgp/platform/osgp-adapter-domain-core/src/test/java/org/opensmartgridplatform/adapter/domain/core/application/services/DefaultDeviceResponseServiceTest.java [hunk 2]
```diff
@@ -0,0 +1,139 @@
...
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.mockito.Mockito.verify;
+import org.opensmartgridplatform.shared.exceptionhandling.ComponentType;
+
+
+import org.junit.jupiter.api.Test;
+
+@ExtendWith(MockitoExtension.class)
+@MockitoSettings(strictness = Strictness.LENIENT)
```
Comment (robindenadel) [inline osgp/platform/osgp-adapter-domain-core/src/test/java/org/opensmartgridplatform/adapter/domain/core/application/services/DefaultDeviceResponseServiceTest.java:25]: Remove this line; the default strictness is better.

File: osgp/platform/osgp-adapter-domain-core/src/test/java/org/opensmartgridplatform/adapter/domain/core/application/services/DefaultDeviceResponseServiceTest.java (hunk 1 above)
Comment (robindenadel) [inline osgp/platform/osgp-adapter-domain-core/src/test/java/org/opensmartgridplatform/adapter/domain/core/application/services/DefaultDeviceResponseServiceTest.java:1]: do not use tabs, check your IDE settings

File: osgp/platform/osgp-adapter-domain-core/src/test/java/org/opensmartgridplatform/adapter/domain/core/application/services/DefaultDeviceResponseServiceTest.java [hunk 3]
```diff
@@ -0,0 +1,139 @@
...
+
+    @Mock
+    private WebServiceResponseMessageSender webServiceResponseMessageSender;
+    
+    @InjectMocks
+    private DefaultDeviceResponseService defaultDeviceResponseService;
+    
+    @Test
+    public void testDefaultDeviceResponseWithNotOkTypeAndException() {
+    	//Arrange
```
Comment (robindenadel) [inline osgp/platform/osgp-adapter-domain-core/src/test/java/org/opensmartgridplatform/adapter/domain/core/application/services/DefaultDeviceResponseServiceTest.java:37]: A single-line comment should be preceded by a blank line, see https://www.oracle.com/java/technologies/javase/codeconventions-comments.html#341

File: osgp/platform/osgp-adapter-domain-core/src/test/java/org/opensmartgridplatform/adapter/domain/core/application/services/DefaultDeviceResponseServiceTest.java [hunk 4]
```diff
@@ -0,0 +1,139 @@
...
+    	CorrelationIds ids = new CorrelationIds("orginazationTestId", "deviceIdTest", "correlationUid");
```
Comment (robindenadel) [inline osgp/platform/osgp-adapter-domain-core/src/test/java/org/opensmartgridplatform/adapter/domain/core/application/services/DefaultDeviceResponseServiceTest.java:38]: new CorrelationIds("orginazationTestId", "deviceIdTest", "correlationUid") is used in more tests, refactor it into a constant CORRELATION_IDS

File: osgp/platform/osgp-adapter-domain-core/src/test/java/org/opensmartgridplatform/adapter/domain/core/application/services/DefaultDeviceResponseServiceTest.java [hunk 5]
```diff
@@ -0,0 +1,139 @@
...
+    	String messageType = "Warning";
+    	int messagePriority = 3;
```
Comment (robindenadel) [inline osgp/platform/osgp-adapter-domain-core/src/test/java/org/opensmartgridplatform/adapter/domain/core/application/services/DefaultDeviceResponseServiceTest.java:40]: refactor these into constants

File: osgp/platform/osgp-adapter-domain-core/src/test/java/org/opensmartgridplatform/adapter/domain/core/application/services/DefaultDeviceResponseServiceTest.java [hunk 6]
```diff
@@ -0,0 +1,139 @@
...
+    	
+    	ArgumentCaptor<ResponseMessage> argument = ArgumentCaptor.forClass(ResponseMessage.class);
+    	verify(webServiceResponseMessageSender).send(argument.capture());
+
+    	//Assert
+		assertThat(argument.getValue()).usingRecursiveComparison().isEqualTo(expectedResponseMessage);
+    }
+    
+    @Test
+    public void testDefaultDeviceResponseWithNotOkTypeAndNoException2() {
```
Comment (robindenadel) [inline osgp/platform/osgp-adapter-domain-core/src/test/java/org/opensmartgridplatform/adapter/domain/core/application/services/DefaultDeviceResponseServiceTest.java:62]: the difference with the previous test should be clear from the method name

**Review by robindenadel — CHANGES_REQUESTED** [review]

File: osgp/platform/osgp-adapter-domain-core/src/test/java/org/opensmartgridplatform/adapter/domain/core/application/services/DefaultDeviceResponseServiceTest.java (hunk 3 above)
Comment (arjen0203) [reply osgp/platform/osgp-adapter-domain-core/src/test/java/org/opensmartgridplatform/adapter/domain/core/application/services/DefaultDeviceResponseServiceTest.java:44]: All single-line comments should now be proceded with a blank line

File: osgp/platform/osgp-adapter-domain-core/src/test/java/org/opensmartgridplatform/adapter/domain/core/application/services/DefaultDeviceResponseServiceTest.java (hunk 1 above)
Comment (arjen0203) [reply osgp/platform/osgp-adapter-domain-core/src/test/java/org/opensmartgridplatform/adapter/domain/core/application/services/DefaultDeviceResponseServiceTest.java:9]: header is now added

Comment (arjen0203) [reply osgp/platform/osgp-adapter-domain-core/src/test/java/org/opensmartgridplatform/adapter/domain/core/application/services/DefaultDeviceResponseServiceTest.java:9]: I should now have properly have set up my IDE settings

File: osgp/platform/osgp-adapter-domain-core/src/test/java/org/opensmartgridplatform/adapter/domain/core/application/services/DefaultDeviceResponseServiceTest.java (hunk 6 above)
Comment (arjen0203) [reply osgp/platform/osgp-adapter-domain-core/src/test/java/org/opensmartgridplatform/adapter/domain/core/application/services/DefaultDeviceResponseServiceTest.java]: Difference should be clear in the name, The added 2 at the end was accidental and added confusion. This has now been removed

File: osgp/platform/osgp-adapter-domain-core/src/test/java/org/opensmartgridplatform/adapter/domain/core/application/services/DefaultDeviceResponseServiceTest.java (hunk 5 above)
Comment (arjen0203) [reply osgp/platform/osgp-adapter-domain-core/src/test/java/org/opensmartgridplatform/adapter/domain/core/application/services/DefaultDeviceResponseServiceTest.java]: The variables that are used in each test method have been refactored into constants

File: osgp/platform/osgp-adapter-domain-core/src/test/java/org/opensmartgridplatform/adapter/domain/core/application/services/DefaultDeviceResponseServiceTest.java [hunk 7]
```diff
@@ -0,0 +1,143 @@
+/**
```
Comment (robindenadel) [inline osgp/platform/osgp-adapter-domain-core/src/test/java/org/opensmartgridplatform/adapter/domain/core/application/services/DefaultDeviceResponseServiceTest.java:1]: Formatting not ok, check your settings.

File: osgp/platform/osgp-adapter-domain-core/src/test/java/org/opensmartgridplatform/adapter/domain/core/application/services/DefaultDeviceResponseServiceTest.java (hunk 1 above)
Comment (robindenadel) [reply osgp/platform/osgp-adapter-domain-core/src/test/java/org/opensmartgridplatform/adapter/domain/core/application/services/DefaultDeviceResponseServiceTest.java:9]: I still see its not ok.

File: osgp/platform/osgp-adapter-domain-core/src/test/java/org/opensmartgridplatform/adapter/domain/core/application/services/DefaultDeviceResponseServiceTest.java [hunk 8]
```diff
@@ -0,0 +1,143 @@
...
+
+public class DefaultDeviceResponseServiceTest {
+
+    @Mock
+    private WebServiceResponseMessageSender webServiceResponseMessageSender;
+    
+    @InjectMocks
+    private DefaultDeviceResponseService defaultDeviceResponseService;
+
+	//constants that are used in all the tests
```
Comment (robindenadel) [inline osgp/platform/osgp-adapter-domain-core/src/test/java/org/opensmartgridplatform/adapter/domain/core/application/services/DefaultDeviceResponseServiceTest.java:36]: remove this comment

File: osgp/platform/osgp-adapter-domain-core/src/test/java/org/opensmartgridplatform/adapter/domain/core/application/services/DefaultDeviceResponseServiceTest.java [hunk 9]
```diff
@@ -0,0 +1,143 @@
...
+	private static final CorrelationIds ids = new CorrelationIds("orginazationTestId", "deviceIdTest", "correlationUid");
+	private static final String messageType = "Warning";
+	private static final int messagePriority = 3;
```
Comment (robindenadel) [inline osgp/platform/osgp-adapter-domain-core/src/test/java/org/opensmartgridplatform/adapter/domain/core/application/services/DefaultDeviceResponseServiceTest.java:39]: Constants should be all uppercase with words separated by underscores, see https://www.oracle.com/java/technologies/javase/codeconventions-namingconventions.html

File: osgp/platform/osgp-adapter-domain-core/src/test/java/org/opensmartgridplatform/adapter/domain/core/application/services/DefaultDeviceResponseServiceTest.java (hunk 1 above)
Comment (robindenadel) [reply osgp/platform/osgp-adapter-domain-core/src/test/java/org/opensmartgridplatform/adapter/domain/core/application/services/DefaultDeviceResponseServiceTest.java:9]: still not ok

**Review by robindenadel — CHANGES_REQUESTED** [review]

File: osgp/platform/osgp-adapter-domain-core/src/test/java/org/opensmartgridplatform/adapter/domain/core/application/services/DefaultDeviceResponseServiceTest.java (hunk 9 above)
Comment (arjen0203) [reply osgp/platform/osgp-adapter-domain-core/src/test/java/org/opensmartgridplatform/adapter/domain/core/application/services/DefaultDeviceResponseServiceTest.java]: These should now be changed to upercase words separated by underscores

File: osgp/platform/osgp-adapter-domain-core/src/test/java/org/opensmartgridplatform/adapter/domain/core/application/services/DefaultDeviceResponseServiceTest.java (hunk 8 above)
Comment (arjen0203) [reply osgp/platform/osgp-adapter-domain-core/src/test/java/org/opensmartgridplatform/adapter/domain/core/application/services/DefaultDeviceResponseServiceTest.java]: comment has been removed

File: osgp/platform/osgp-adapter-domain-core/src/test/java/org/opensmartgridplatform/adapter/domain/core/application/services/DefaultDeviceResponseServiceTest.java (hunk 1 above)
Comment (arjen0203) [reply osgp/platform/osgp-adapter-domain-core/src/test/java/org/opensmartgridplatform/adapter/domain/core/application/services/DefaultDeviceResponseServiceTest.java:9]: I now manually ran the auto code format in the eclipse workspace. If it now isn't correct I'll need to have a look at my eclipse set up.

**Comment by robindenadel** [comment]
retest this please

**Review by robindenadel — CHANGES_REQUESTED** [review]

**Review by robindenadel — APPROVED** [review]

**Pull request merged**